test/DeltaTriplesTest.cpp

```diff
@@ -449,6 +450,139 @@ TEST_F(DeltaTriplesTest, DeltaTriplesManager) {
+  DeltaTriplesManager deltaTriplesManager(testQec->getIndex().getImpl());
+  auto& vocab = testQec->getIndex().getVocab();
+  auto cancellationHandle =
+      std::make_shared<ad_utility::CancellationHandle<>>();
+
+  LocalVocab localVocab;
+
+  auto insertTriples = [&, this](const std::vector<std::string>& turtles) {
+    auto triples = makeIdTriples(vocab, localVocab, turtles);
+    auto updateSnapshot =
+        getRuntimeParameter<&RuntimeParameters::propagateChangesFromUpdates_>();
+    deltaTriplesManager.modify<void>(
+        [&](DeltaTriples& deltaTriples) {
+          deltaTriples.insertTriples(cancellationHandle, std::move(triples));
+        },
+        {.updateSnapshotAfterRequest_ = updateSnapshot});
+  };
+
+  // Initially the changes from updates should be propagated.
+  EXPECT_TRUE(
+      getRuntimeParameter<&RuntimeParameters::propagateChangesFromUpdates_>());
+  auto initialSnapshot = deltaTriplesManager.getCurrentSnapshot();
+  auto initialIndex = initialSnapshot->index_;
+
+  insertTriples({"<A> <B> <C>"});
+
+  auto snapshotAfterInsert = deltaTriplesManager.getCurrentSnapshot();
+  EXPECT_NE(snapshotAfterInsert->index_, initialIndex);
+
+  // Disable the propagation of the changes from updates.
+  setRuntimeParameter<&RuntimeParameters::propagateChangesFromUpdates_>(false);
+
+  auto snapshotBeforeSecondUpdate = deltaTriplesManager.getCurrentSnapshot();
+  auto indexBeforeSecondUpdate = snapshotBeforeSecondUpdate->index_;
+  insertTriples({"<D> <E> <F>"});
+  auto snapshotAfterSecondInsert = deltaTriplesManager.getCurrentSnapshot();
+  EXPECT_EQ(snapshotAfterSecondInsert->index_, indexBeforeSecondUpdate);
+
+  // Force snapshot creation should work regardless of parameter
+  deltaTriplesManager.updateStoredSnapshot();
+  auto snapshotAfterForce = deltaTriplesManager.getCurrentSnapshot();
+  EXPECT_NE(snapshotAfterForce->index_, indexBeforeSecondUpdate);
+
+  // The changes from the updates should directly be visible after enabling.
+  setRuntimeParameter<&RuntimeParameters::propagateChangesFromUpdates_>(true);
+
+  auto snapshotBeforeThirdUpdate = deltaTriplesManager.getCurrentSnapshot();
+  auto indexBeforeThirdUpdate = snapshotBeforeThirdUpdate->index_;
+  EXPECT_NE(indexBeforeThirdUpdate, snapshotAfterForce->index_);
+
+  insertTriples({"<G> <H> <I>"});
+
+  auto snapshotAfterThirdInsert = deltaTriplesManager.getCurrentSnapshot();
+  EXPECT_NE(snapshotAfterThirdInsert->index_, indexBeforeThirdUpdate);
+}
+
+// _____________________________________________________________________________
+TEST_F(DeltaTriplesTest, propagateChangesFromUpdatesMetadataBehavior) {
+  DeltaTriplesManager deltaTriplesManager(testQec->getIndex().getImpl());
+  auto& vocab = testQec->getIndex().getVocab();
+  auto cancellationHandle =
+      std::make_shared<ad_utility::CancellationHandle<>>();
+
+  LocalVocab localVocab;
+
+  auto setPropagateChangesFromUpdates =
+      setRuntimeParameter<&RuntimeParameters::propagateChangesFromUpdates_,
```

Comment:
We have a helper function (I don't recall the name exactly but it is something along the lines of `setRuntimeParameterForTesting`) that does also restore the original value of the runtime parameter at the end of the scope. This way you won't get any weird interference across tests if some test depends on the default value.